### PR TITLE
Migrate Rely to Cram tests

### LINKED ISSUE #15: move test from rely to dune cram tests

### LINKED ISSUE #20: run generated files through fmt
Any default settings would be better than no formatting. Although a real win would be to check for a local `.ocamlformat` file.

### PR DESCRIPTION
**Pull request by naartjie**
fixes #15 
fixes #20

### CODE REVIEW

File: test/custom_output_prefix.t/run.t [hunk 1]
```diff
@@ -0,0 +1,2183 @@
+  $ dune exec -- stork gen -o custom_output_prefix original_output_prefix_1.atd original_output_prefix_bar.atd
```
Comment (wokalski) [inline test/custom_output_prefix.t/run.t:1]: I love this!! Although maybe let's split it into separate files? What do you think? This is quite overwhelming.

**Comment by wokalski** [comment]
The tests look great although they are a bit overwhelming. Too many changes in one file I think. Can we test different files in different cram tests? I.e. `cat <*>.mli` in a different file than `cat <*>.ml` ?

**Comment by naartjie** [comment]
> The tests look great although they are a bit overwhelming. Too many changes in one file I think. Can we test different files in different cram tests? I.e. `cat <*>.mli` in a different file than `cat <*>.ml` ?

I thought about this same thing when implementing this, and I had a bit of a dilemma (thought there would be too many files). 

I had a stab at it [here](https://github.com/dialohq/stork/commit/121c8e5b05d3c6091a01d735b85c1ef725366cce) (it fixes #20 at the same time). Let me know what you think, and if that's what you had in mind.

**Comment by wokalski** [comment]
@naartjie you can use run.t in directory tests:
https://github.com/ocaml/dune/tree/main/example/hello_world.t

It's a bit confusing but  basically you create a  directory ending with .t and then you create run.t inside that directory.

Then all files generated by that command can be simply committed to that directory. It's quite cool.

**Comment by naartjie** [comment]
> you can use run.t in directory tests

@wokalski how do you want to use directory tests? Should each directory in `./test/` become a mini dune project, where we use `run.t` to invoke the tests? Or are you thinking in some other direction?

**Comment by naartjie** [comment]
@wokalski this is ready for review again. I have split up all the cramtests into separate files

File: test/test_utils.ml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  let json_string = Yojson.(Safe.from_file file |> Yojson.Safe.to_string) in
+  let doc_of_string = from_string json_string in
+  let res_of_string = to_string doc_of_string in
+  let doc_read = read_from_file file in
+  let res_read = to_string doc_read in
+  assert (res_of_string = res_read);
+  print_endline res_read;
+  print_endline "---";
+  print_endline res_of_string;
+  print_endline "---"
```
Comment (wokalski) [inline test/test_utils.ml:11]: Very good idea @naartjie 😄

**Review by wokalski — APPROVED** [review]

**Pull request merged**